iothub/device/src/Edge/EdgeModuleClientHelper.cs

```diff
@@ -103,27 +71,65 @@ private async Task<IotHubModuleClient> CreateInternalClientFromEnvironmentAsync(
 
                 Debug.WriteLine("EdgeModuleClientFactory setupTrustBundle from service");
 
-                ICertificateValidator certificateValidator = NullCertificateValidator.Instance;
+                return new IotHubConnectionCredentials(authMethod, hostName, gateway);
+            }
+        }
+
+        internal static async Task<ICertificateValidator> CreateCertificateValidatorFromEnvironmentAsync(ITrustBundleProvider trustBundleProvider, IotHubClientOptions options)
+        {
+            // Make sure client options is initialized.
+            if (options == default)
```

Comment:
Can we assume that by now options will have an instance? I'd be fine with a `Debug.Assert()` too.